app/services/policies/policies_list_service.rb

```diff
@@ -1,11 +1,59 @@
 # frozen_string_literal: true
 
 require 'jsonclient'
+require 'set'
 
 class Policies::PoliciesListService
-
-  def self.call
+  def self.call(account)
     response = ::JSONClient.get(ThreeScale.config.sandbox_proxy.apicast_registry_url)
-    response.body.symbolize_keys[:policies] if response.ok?
+    list = response.ok? ? PolicyList.from_hash(response.body['policies']) : PolicyList.new
+    list.merge!(PolicyList.new(account.policies)) if account.provider_can_use?(:policy_registry)
+    list.to_h
+  end
+
+  class PolicyList
+    include Enumerable
```

Comment:
I included this but it is not used at the moment. Only because I was doing some testing in the console and I did not have access to `@sets` which is protected (on purpose)